mongo/crud_examples_test.go

```diff
@@ -0,0 +1,525 @@
+}
+
+func ExampleCollection_CountDocuments() {
+	var coll *mongo.Collection
+
+	// count the number of times the name "Bob" appears in the collection
+	// specify the MaxTime option to limit the amount of time the operation can run on the server
+	opts := options.Count().SetMaxTime(2 * time.Second)
+	count, err := coll.CountDocuments(context.TODO(), bson.D{{"name", "Bob"}}, opts)
+	if err != nil {
+		log.Fatal(err)
+	}
+	fmt.Printf("name Bob appears in %v documents", count)
+}
+
+func ExampleCollection_DeleteMany() {
+	var coll *mongo.Collection
+
+	// delete all documents in which the "name" field is "Bob" or "bob"
+	// specify the Collation option to provide a collation that will ignore case for string comparisons
+	opts := options.Delete().SetCollation(&options.Collation{
+		Locale:   "en_US",
+		Strength: 1,
+	})
+	res, err := coll.DeleteMany(context.TODO(), bson.D{{"name", "bob"}}, opts)
+	if err != nil {
+		log.Fatal(err)
+	}
+	fmt.Printf("deleted %v documents\n", res.DeletedCount)
+}
+
+func ExampleCollection_DeleteOne() {
+	var coll *mongo.Collection
+
+	// delete at most one document in which the "name" field is "Bob" or "bob"
+	// specify the SetCollation option to provide a collation that will ignore case for string comparisons
+	opts := options.Delete().SetCollation(&options.Collation{
+		Locale:   "en_US",
+		Strength: 1,
+	})
+	res, err := coll.DeleteOne(context.TODO(), bson.D{{"name", "bob"}}, opts)
+	if err != nil {
+		log.Fatal(err)
+	}
+	fmt.Printf("deleted %v documents\n", res.DeletedCount)
+}
+
+func ExampleCollection_Distinct() {
+	var coll *mongo.Collection
+
+	// find all unique values for the "name" field for documents in which the "age" field is greater than 25
+	// specify the MaxTime option to limit the amount of time the operation can run on the server
+	filter := bson.D{{"age", bson.D{{"$gt", 25}}}}
+	opts := options.Distinct().SetMaxTime(2 * time.Second)
+	values, err := coll.Distinct(context.TODO(), "name", filter, opts)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	for _, value := range values {
+		fmt.Println(value)
+	}
+}
+
+func ExampleCollection_EstimatedDocumentCount() {
+	var coll *mongo.Collection
+
+	// get and print an estimated of the number of documents in the collection
+	// specify the MaxTime option to limit the amount of time the operation can run on the server
+	opts := options.EstimatedDocumentCount().SetMaxTime(2 * time.Second)
+	count, err := coll.EstimatedDocumentCount(context.TODO(), opts)
+	if err != nil {
+		log.Fatal(err)
+	}
+	fmt.Printf("estimated document count: %v", count)
+}
+
+func ExampleCollection_Find() {
+	var coll *mongo.Collection
+
+	// find all documents in which the "name" field is "Bob"
+	// specify the Sort option to sort the returned documents by name in ascending order
+	opts := options.Find().SetSort(bson.D{{"name", 1}})
+	cursor, err := coll.Find(context.TODO(), bson.D{{"name", "Bob"}}, opts)
```

Comment:
It doesn't make a lot of sense to filter on a single value of a field, and then also sort by the same field.  Perhaps sort by something more specific than name, e.g. nationalId.

Reply:
Yeah this was supposed to sort by age. Changed the sort document to `{age: 1}`.